src/lib/components/multiselect/MultiSelect.component.js

```diff
@@ -0,0 +1,187 @@
+//@flow
+import React from "react";
+import styled from "styled-components";
+import * as defaultTheme from "../../style/theme";
+import CheckBox from "./../checkbox/Checkbox.component";
+import Button from "./../button/Button.component";
+import Select from "./../select/Select.component";
+
+export type ItemProps = {
+  selected?: boolean,
+  isFavorite?: boolean,
+  label: string,
+  description?: string,
+  onSelect?: (any, any) => void, //on checkbox click
+  onItemRemove?: (any, any) => void, //on item remove button click
+  onFavoriteClick?: (any, any) => void // on Favorite icon click
+};
+
+export type OptionProps = {
+  label: string,
+  value: string | number
+};
+
+export type SearchProps = {
+  placeholder?: string,
+  options: Array<OptionProps>, // "Content of option"
+  onSelect: any => void, //on option click
+  onAdd?: any => void,
+  selectedOption: any
+};
+
+type MultiSelectProps = {
+  title?: string,
+  items: Array<ItemProps>,
+  search?: SearchProps,
+  onItemRemove?: (any, any) => void //on item remove button click
+};
+
+const MultiSelectContainer = styled.div`
+  padding: ${defaultTheme.padding.base};
+`;
+
+const MultiSelectTitle = styled.h3`
+  color: ${defaultTheme.grayDarker};
+  font-weight: ${defaultTheme.fontWeight.bold};
+  font-size: ${defaultTheme.fontSize.large};
+`;
+
+const MultiSelectItemContainer = styled.div`
+  margin: ${defaultTheme.padding.smaller} 0;
+  padding: ${defaultTheme.padding.small} 0;
+  display: flex;
+  align-items: center;
+  border-bottom: 1px solid gray;
+  &:last-child {
+    border: none;
+  }
+`;
+
+const MultiSelectSearchContainer = styled.div`
+  display: flex;
+  .sc-select-container {
+    width: 100%;
+  }
+  .sc-button {
+    margin: 0 ${defaultTheme.padding.small};
+  }
+`;
+
+const MultiSelectItemLeft = styled.div`
+  .sc-checkbox,
+  .sc-button {
+    margin: 0 ${defaultTheme.padding.small};
+  }
+`;
+const MultiSelectItemCenter = styled.div`
+  display: flex;
+  flex-grow: 1;
+  align-items: center;
+  justify-content: space-between;
+`;
+const MultiSelectItemRight = styled.div`
+  padding: 0 ${defaultTheme.padding.base};
+`;
+
+const MultiSelectItemLabel = styled.span`
+  font-size: ${defaultTheme.fontSize.large};
+`;
+
+const MultiSelectItemDescription = styled.span``;
+
+function MultiSelectItem(props: ItemProps) {
+  const {
+    selected,
+    label,
+    description,
+    onItemRemove,
+    onSelect,
+    isFavorite,
+    onFavoriteClick
+  } = props;
+
+  return (
+    <MultiSelectItemContainer className="sc-multi-select-item">
+      <MultiSelectItemLeft className="sc-multi-select-item-left">
+        {typeof selected !== "undefined" && onSelect && (
+          <CheckBox
+            checked={selected}
+            onChange={event => onSelect(label, event)}
+          />
+        )}
+        {typeof isFavorite !== "undefined" && onFavoriteClick && (
```

Comment:
Same as `selected` but in this case should we rename it to `isSelected` for consistency ?